components/LanguageSwitcher.jsx

```diff
@@ -10,6 +10,15 @@ const LanguageSwitchLink = ({ locale, ...rest }) => {
   const theme = useTheme()
   const { t } = useTranslation('common')
 
+  const handleLanguageChange = (newLocale) => {
+    localStorage.setItem('appLocale', newLocale);
```

Comment:
No need to introduce another KV pair in localStorage, as currently a KV pair for the locale exists, even though there is no place in the code that sets and gets the localStorage value, the package _next-i18next_ does this automatically. The key, is called i18nextLng and you can work with that. Also feel free to test it's functionality, it's the same as the appLocale, that you introduce.
![LocalStorage](https://github.com/TUM-Dev/TUMenu/assets/99186919/b6a27fe3-89e3-433b-85f9-fe773b30c983)

Reply:
Fixed in https://github.com/TUM-Dev/TUMenu/pull/110/commits/972b7d8ca37be9e33adeda3e0367e6b305a155f4